frontend/packages/core/src/Input/text-field.tsx

```diff
@@ -162,6 +208,55 @@ export const TextField = ({
     );
   }
 
+  if (isAutoCompleteable) {
+    const [autoCompleteOptions, setAutoCompleteOptions] = React.useState([]);
+    const autoCompleteDebounce = React.useRef(
+      _.debounce(value => {
+        autocompleteCallback(value)
+          .then(data => {
+            setAutoCompleteOptions(data.results);
+          })
+          .catch(err => {
+            helpText = err;
+          });
+      }, 500)
+    ).current;
+
+    // TODO (mcutalo): support option.label in the renderOption
+    return (
+      <Autocomplete
+        freeSolo
+        size="small"
+        options={autoCompleteOptions}
+        PopperComponent={renderPopper}
+        getOptionLabel={option => (option.id ? option.id : option)}
+        onInputChange={(e, v, r) => autoCompleteDebounce(v)}
+        renderOption={option => (
+          <ResultGrid container alignItems="center">
+            <Grid item xs>
+              <ResultLabel>{option.id}</ResultLabel>
+            </Grid>
+          </ResultGrid>
+        )}
+        renderInput={params => (
+          <StyledTextField
+            {...params}
```

Comment:
reactism
```suggestion
        renderInput={props => (
          <StyledTextField
            {...props}
```

Reply:
Either way works, i can use this name also.